intrastat_delivery/models/sale_order.py

```diff
@@ -8,12 +8,15 @@ class SaleOrder(models.Model):
     def _action_confirm(self):
         ret = super()._action_confirm()
         for order in self:
-            order.write(
-                {
-                    "incoterm": order.carrier_id.incoterm.id,
-                    "intrastat_transport_id": order.carrier_id.intrastat_transport_id.id,
-                }
-            )
+            vals = {}
+            if order.carrier_id.incoterm:
```

Comment:
@nguyenminhchien this would be incorrect in the case:
- at time `t`, user confirms the SO, which the carrier has defined incoterm
- at time `t+1`, user removes the incoterm on carrier, set SO to draft, and re-confirm it
  The SO will keep the previous incoterm on the SO

Reply:
i got it, but if keep as before: always update `vals["incoterm"] = order.carrier_id.incoterm.id`. it does not respect what written in the README:

![image](https://github.com/nguyenminhchien/intrastat-extrastat/assets/8724793/e40abe46-45e0-4ffa-836b-2a8604eccb82)

Do you have any suggestion here?

Reply:
nice catch, if so, that will be a fix in a seperated commit